provision.sh

```diff
@@ -14,31 +14,47 @@ echo "---------------------------------------------"
 echo "Running vagrant provisioning"
 echo "---------------------------------------------"
 
+# install heroku toolbelt
+echo "-------------- Installing heroku toolbelt -------------------------"
+#wget -O- https://toolbelt.heroku.com/install-ubuntu.sh | sh
+# These shell script snippets are directly taken from heroku installation script
+# We want to avoid the apt-get update
+# add heroku repository to apt
+echo "deb http://toolbelt.heroku.com/ubuntu ./" > /etc/apt/sources.list.d/heroku.list
+# install heroku's release key for package verification
+wget -O- https://toolbelt.heroku.com/apt/release.key | apt-key add -
```

Comment:
The key addition was part of earlier install. It's standard practice to do a key-based verification that what you are installing is indeed what it claims to be.

Check `rvm` (Ruby version manager) install for example. It starts with adding a key to your machine so that you can verify it.

Similar stuff is also done under the hood by package managers or build tools like Maven for Java, or pip for Python.